src/main/java/at/hannibal2/skyhanni/features/inventory/ItemPickupLog.kt

```diff
@@ -0,0 +1,341 @@
+    private var itemsRemovedFromInventory = mutableMapOf<Int, PickupEntry>()
+    private var display: Renderable? = null
+
+    private val patternGroup = RepoPattern.group("itempickuplog")
+    private val shopPattern by patternGroup.pattern(
+        "shoppattern",
+        "^(?<itemName>.+?)(?: x\\d+)?\$",
+    )
+
+    private val bannedItemsPattern by patternGroup.list(
+        "banneditems",
+        "SKYBLOCK_MENU",
+        "CANCEL_PARKOUR_ITEM",
+        "CANCEL_RACE_ITEM",
+    )
+    private val bannedItemsConverted = bannedItemsPattern.map { it.toString().asInternalName() }
+
+    @SubscribeEvent
+    fun onRenderOverlay(event: GuiRenderEvent) {
+        if (!isEnabled()) return
+        display?.let { config.pos.renderRenderable(it, posLabel = "Item Pickup Log Display") }
+    }
+
+    @SubscribeEvent
+    fun onWorldChange(event: LorenzWorldChangeEvent) {
+        if (!isEnabled()) return
+        itemList.clear()
+        itemsAddedToInventory.clear()
+        itemsRemovedFromInventory.clear()
+    }
+
+    @SubscribeEvent
+    fun onSackChange(event: SackChangeEvent) {
+        if (!isEnabled() || !config.sack) return
+
+        event.sackChanges.forEach {
+            val itemStack = (it.internalName.getItemStack())
+            val item = PickupEntry(itemStack.dynamicName(), it.delta.absoluteValue.toLong(), it.internalName)
+
+            updateItem(itemStack.hash(), item, itemStack, it.delta < 0)
+        }
+    }
+
+    @SubscribeEvent
+    fun onPurseChange(event: PurseChangeEvent) {
+        if (!isEnabled() || !config.coins || !worldChangeCooldown()) return
+
+        updateItem(0, PickupEntry("§6Coins", event.coins.absoluteValue.toLong(), coinIcon), coinIcon.getItemStack(), event.coins < 0)
+    }
+
+    @SubscribeEvent
+    fun onTick(event: LorenzTickEvent) {
+        if (!isEnabled()) return
+
+        val oldItemList = mutableMapOf<Int, Pair<ItemStack, Int>>()
+
+        oldItemList.putAll(itemList)
+
+        itemList.clear()
+
+        val inventoryItems = InventoryUtils.getItemsInOwnInventory().toMutableList()
+        val cursorItem = Minecraft.getMinecraft().thePlayer.inventory?.itemStack
+
+        if (cursorItem != null) {
+            val hash = cursorItem.hash()
+            //this prevents items inside hypixel guis counting when picked up
+            if (oldItemList.contains(hash)) {
+                inventoryItems.add(cursorItem)
+            }
+        }
+
+        for (itemStack in inventoryItems) {
+            val hash = itemStack.hash()
+            val old = itemList[hash]
+            if (old != null) {
+                itemList[hash] = old.copy(second = old.second + itemStack.stackSize)
+            } else {
+                itemList[hash] = itemStack to itemStack.stackSize
+            }
+        }
+
+        if (!worldChangeCooldown()) return
+
+        checkForDuplicateItems(itemList, oldItemList, false)
+        checkForDuplicateItems(oldItemList, itemList, true)
+
+        updateDisplay()
```

Comment:
make it so that update display only gets called when the display changes, not every tick.

Reply:
are you sure you want this? i could do it but i would have to save an entire list of every item and check the next list against it every tick which i think could end out being more inefficent than just updating the display every tick. if you see a better way please tell me <3

Reply:
done